api/src/controllers/project-invite.controller.ts

```diff
@@ -0,0 +1,136 @@
+import { Controller, UseGuards, Get, Req, Param, Inject, Post, Body, Delete, Patch } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { InjectRepository } from '@nestjs/typeorm';
+import { User } from 'entity/user.entity';
+import AuthorizationService from 'services/authorization.service';
+import MailService from 'services/mail.service';
+import { Repository } from 'typeorm';
+import { ProjectAction } from 'domain/actions';
+import { Invite, InviteStatus } from 'entity/invite.entity';
+import { InviteUserRequest, UpdateProjectInviteRequest } from 'domain/http';
+import { ProjectUser } from 'entity/project-user.entity';
+
+@Controller('api/v1/projects')
+@UseGuards(AuthGuard())
+export default class ProjectInviteController {
+  constructor(
+    private auth: AuthorizationService,
+    private mail: MailService,
+    @InjectRepository(User) private userRepo: Repository<User>,
+    @InjectRepository(Invite) private inviteRepo: Repository<Invite>,
+    @InjectRepository(ProjectUser) private projectUserRepo: Repository<ProjectUser>,
+  ) {}
+
+  @Get(':projectId/invites')
+  async find(@Req() req, @Param('projectId') projectId: string) {
+    const user = this.auth.getRequestUserOrClient(req, { mustBeUser: true });
+    await this.auth.authorizeProjectAction(user, projectId, ProjectAction.ViewProjectUsers);
+
+    const invites = await this.inviteRepo.find({
+      where: { project: { id: projectId }, status: InviteStatus.Sent },
+    });
+
+    return {
+      data: invites,
+    };
+  }
+
+  @Post(':projectId/invites')
+  async create(@Req() req, @Param('projectId') projectId: string, @Body() inviteUserRequest: InviteUserRequest) {
+    const user = this.auth.getRequestUserOrClient(req, { mustBeUser: true });
+    await this.auth.authorizeProjectAction(user, projectId, ProjectAction.AddProjectUser);
```

Comment:
We should probably rename this ProjectAction to `InviteProjectUser` for consistency and Remove the `AddProjectUser` action since it's not supported other than via invites. Just be careful to refactor all the existing usages also in the frontend.

Reply:
makes sense